server/src/main/java/org/opensearch/gateway/TransportNodesListGatewayStartedShardsBatch.java

```diff
@@ -147,16 +149,19 @@ protected NodeGatewayStartedShardsBatch nodeOperation(NodeRequest request) {
                         namedXContentRegistry,
                         nodeEnv,
                         indicesService,
-                        shardAttr.getCustomDataPath(),
+                        shardAttr.getValue().getCustomDataPath(),
                         settings,
                         clusterService
                     )
                 );
             } catch (Exception e) {
-                shardsOnNode.put(
-                    shardId,
-                    new GatewayStartedShard(null, false, null, new OpenSearchException("failed to load started shards", e))
-                );
+                // should return null in case of known exceptions being returned from getShardInfoOnLocalNode method.
+                if (e instanceof IllegalStateException || e.getMessage().contains(INDEX_NOT_FOUND) || e instanceof IOException) {
+                    shardsOnNode.put(shardId, null);
+                } else {
+                    // return actual exception as it is for unknown exceptions
+                    shardsOnNode.put(shardId, new GatewayStartedShard(null, false, null, e));
```

Comment:
it was translated to OpenSearchException before, lets do the same?

Reply:
Shard level exception which is supposed to be stored in the cache was never translated to OpenSearchException. Sp, keeping it as it is. Exceptions which were supposed to ignored, those were wrapped with `OpenSearchException`, but for those, now we're sending null.